src/index.ts

```diff
@@ -174,8 +174,9 @@ export class LinkChecker extends EventEmitter {
     let state = LinkState.BROKEN;
     let data = '';
     let shouldRecurse = false;
+    let res: gaxios.GaxiosResponse<string> | undefined = undefined;
```

Comment:
No need to do the ` | undefined = undefined` bit here.  You can delete that, and it would work fine!